src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx

```diff
@@ -64,3 +64,26 @@ export function useComponentState() {
 
   return { formik, deleteCourse };
 }
+
+export function useCoursesState() {
```

Comment:
don't create a new hook here. just create a function which responsible for fetch courses. Also define api functions in the api.ts file.